blenderproc/python/types/MeshObjectUtility.py

```diff
@@ -531,13 +531,17 @@ def mesh_as_trimesh(self) -> Trimesh:
         # get mesh data
         mesh = self.get_mesh()
 
-        # get vertices and faces
+        # get vertices 
         verts = np.array([[v.co[0], v.co[1], v.co[2]] for v in mesh.vertices])
-        faces = np.array([[f.vertices[0], f.vertices[1], f.vertices[2]] for f in mesh.polygons])
+        
+        # get faces
+        if len(mesh.polygons[0].vertices[:]) == 4:
+            faces = np.array([f.vertices[:] for f in mesh.polygons if len(f.vertices[:]) == 4])
+        else:
+            faces = np.array([[f.vertices[0], f.vertices[1], f.vertices[2]] for f in mesh.polygons])
```

Comment:
Why not directly always use:

```suggestion
faces = np.array([f.vertices[:] for f in mesh.polygons if len(f.vertices[:]) in [3, 4]])
```

or even without the if

Reply:
The problem is sometimes meshes have both triangles and quads. In that case the Trimesh constructor crashes as it can only take a list of Mx3 or Mx4 faces.

Reply:
My solution isn't great either, I am basically just loading quads if the mesh has both quads and triangles.

Reply:
No that is indeed a problem. 
But wouldnt it be then better to first check that all faces are either quads or triangles and otherwise throw an error?
In your code some faces might just get filtered out, so you would get an incomplete mesh back without knowing about it.

Reply:
Yes, that is true. Hmm I am just not sure how common it is to have meshes with mixed quads / triangles (I have on right now :D). But I agree, we can throw an error and ask the user to triangulate the quads. Doing this by ourselves is a bit of an overkill.